Digitigrade Paralysis BugFix

SpaceDragon00 opened the pull request:
## About The Pull Request

- A limb's attach_limb proc now uses the set_owner() proc instead of setting the owner manually, fixing problems where limbs that were meant to be paralyzed were not
- Added a sanity check to set_usable_legs to ensure the value isn't set lower than 0

## Why It's Good For The Game

Fewer bugs, more sanity

## Changelog
:cl:
fix: People with digitigrade legs are now properly paralyzed by the Paraplegic perk
fix: Usable_limbs can no longer be set below 0, which let people walk when they shouldn't be able to
/:cl:

SpaceDragon00 committed:
Fixing Digitigrade Legs, Adding Sanity Checks

- A limb's attach_limb proc now uses the set_owner() proc instead of setting the owner manually, fixing problems where limbs that were meant to be paralyzed were not
- Added a sanity check to set_usable_legs to ensure the value isn't set lower than 0

LemonInTheDark commented on code/modules/surgery/bodyparts/dismemberment.dm:
```diff
@@ -323,7 +323,7 @@
 		return FALSE
 	. = TRUE
 	moveToNullspace()
-	owner = new_limb_owner
```
Ah this was my fault. wonderful

LemonInTheDark commented on code/modules/mob/living/living.dm:
```diff
@@ -1819,6 +1819,9 @@
 /mob/living/proc/set_usable_legs(new_value)
 	if(usable_legs == new_value)
 		return
+	if(new_value < 0) // Sanity check
+		new_value = 0
+
```
Don't you think we should error on this?

SpaceDragon00 replied:
Good idea, I'll add that real quick

MrMelbert commented:
Fixes #56141

SpaceDragon00 committed:
Adds A Stack Trace to Set_Usable_Legs

- set_usable_legs now calls stack_trace() whenever it is provided with a negative value

Time-Green commented:
There go the days of having my legs replaced to walk again
